pandas/tests/frame/test_timeseries.py

```diff
@@ -601,3 +601,35 @@ def test_frame_to_period(self):
         tm.assert_index_equal(pts.columns, exp.columns.asfreq('M'))
 
         pytest.raises(ValueError, df.to_period, axis=2)
+
+
+class TestDataFrameTimestamps(object):
+    def test_map_box_timestamps(self):
+        # GH#2689, GH#2627
+        s = Series(date_range('1/1/2000', periods=10))
+
+        def f(x):
+            return (x.hour, x.day, x.month)
+
+        # it works!
+        s.map(f)
```

Comment:
split this test: 1 for series (move to series dir); leave other here but i think it goes in ops
or wherever apply tests are